var/spack/repos/builtin/packages/cp2k/package.py

```diff
@@ -763,3 +824,104 @@ def check(self):
         with spack.util.environment.set_env(CP2K_DATA_DIR=data_dir, PWD=self.build_directory):
             with working_dir(self.build_directory):
                 make("test", *self.build_targets)
+
+
+class CMakeBuilder(spack.build_systems.cmake.CMakeBuilder):
+    """Use the new cmake build system to build cp2k. It is the default when
+    building the master branch of cp2k."""
+
+    def cmake_args(self):
+        spec = self.spec
+
+        if "+cuda" in spec and len(spec.variants["cuda_arch"].value) > 1:
+            raise InstallError("CP2K supports only one cuda_arch at a time")
+
+        if "+rocm" in spec and len(spec.variants["amdgpu_target"].value) > 1:
+            raise InstallError("CP2K supports only one amdgpu_arch at a time")
+
+        args = []
+
+        if spec.satisfies("+cuda") or spec.satisfies("+rocm"):
+            if spec.satisfies("cuda_target"):
+                args += [self.define("CP2K_WITH_GPU", gpu_map[spec.variants["cuda_arch"].value])]
+
+            if spec.satisfies("amdgpu_target"):
+                args += [
+                    self.define("CP2K_WITH_GPU", gpu_map[spec.variants["amdgpu_target"].value])
+                ]
+
+        if spec.satisfies("+cuda"):
+            args += ["-DCP2K_USE_ACCEL=cuda"]
+
+        if spec.satisfies("+rocm"):
+            args += ["-DCP2K_USE_ACCEL=hip"]
+
+        args += [
+            self.define_from_variant("CP2K_ENABLE_REGTESTS", "enable_regtests"),
+            self.define_from_variant("CP2K_USE_ELPA", "elpa"),
+            self.define_from_variant("CP2K_USE_LIBINT2", "libint"),
+            self.define_from_variant("CP2K_USE_SIRIUS", "sirius"),
+            self.define_from_variant("CP2K_USE_SPLA", "spla"),
+            self.define_from_variant("CP2K_USE_COSMA", "cosma"),
+            self.define_from_variant("CP2K_USE_LIBXC", "libxc"),
+            self.define_from_variant("CP2K_USE_LIBTORCH", "pytorch"),
+            self.define_from_variant("CP2K_USE_METIS", "pexsi"),
+            self.define_from_variant("CP2K_USE_SUPERLU", "pexsi"),
+            self.define_from_variant("CP2K_USE_PLUMED", "plumed"),
+            self.define_from_variant("CP2K_USE_SPGLIB", "spglib"),
+            self.define_from_variant("CP2K_USE_VORI", "libvori"),
+            self.define_from_variant("CP2K_USE_SPLA", "spla"),
+            self.define_from_variant("CP2K_USE_QUIP", "quip"),
+        ]
+
+        # we force the use elpa openmp threading support. might need to be revisited though
+        if (spec.satisfies("+openmp") and spec.satisfies("+elpa")) or spec.satisfies(
+            "^elpa+openmp"
+        ):
+            args += ["-DCP2K_ENABLE_ELPA_OPENMP_SUPPORT=ON"]
```

Comment:
This and the the other args below could be rewritten as
```suggestion
        args += [self.define("CP2K_ENABLE_ELPA_OPENMP_SUPPORT",
                 (spec.satisfies("+openmp") and spec.satisfies("+elpa")) or
                 spec.satisfies("^elpa+openmp"))]
```
but this is a matter of taste.